patterns/1-initial/secure-innersource.md

```diff
@@ -0,0 +1,144 @@
+- “Closed Source” is the default in the organization when creating new repositories i.e. only the team owning/maintaining the code has access to the given repo.
+- “Shared Source” within the organization isn’t common practice. Organizational teams aren’t familiar with what code or information should or shouldn’t be placed in shared repositories.
+
+# Forces
+
+- A successful InnerSource program needs openness and transparency (e.g. access to code, issues, documentation, and roadmap)
+- Good security practice is to minimize access, following the [principle of least privilege][principle-of-least-privilege].
+- The Security Team is more familiar with security principles at conceptual level than security practices at operation level. Some of the security policy is good at concept but hard to execute at operation level.
+- Engineering Teams focus more on service/product development or knowledge sharing than on measures for data protection. While it is easy for Engineering Teams to decide to close or open the repository, they are usually not willing to spend time on judging how to reach a balance between both.
+- No one-fit-all guideline or rules to judge what data or process is secured or not. That much depends on data sensitivity and overall security policy/infrastructure.
+
+# Solutions
+
+To reduce the misalignment and possible misunderstanding between the teams involved, it is key to bring everybody together so that they are able to express their goals and concerns, and to develop shared language together.
+
+After that they can decide as a group how to drive the execution of the specific goals, and what to do to reduce the risks that were identified.
+
+## Setup
+
+Start by bringing the Security Team, Engineering Teams and ISPO together, to discuss the goals of the InnerSource initiative, as well as any security concerns.
+
+Some helpful practices are:
+
+- Form an InnerSource task force with representatives of Security, Engineering, and the ISPO to align on the goals of InnerSource adoption.
+- Start with a limited scope to lower the potential security impact caused by the adoption.
+- List the security concerns and figure out mitigation plans together within the task force.
+- Clearly define the different sharing levels of repositories that you want to use in your organization *(Note: A more elaborate model is described under **Extension: An Additional Sharing Level**)*:
+    - **PUBLIC** - open source: accessible for all SW developers in the world
+    - **SHARED** - inner source: accessible for all SW developers in the company
+    - **CLOSED** - closed source: only accessible to named individuals in the company
+
+## Execution
+
+How to allow for a greater amount of SHARED repositories in the organizations depends a lot on the specific business domain, related regulations, and concerns identified in the initial meetings of the InnerSource task force as mentioned above.
+
```

Comment:
Consider adding:

**TIP:** Separate highly specific and differentiating code (the 'secret sauce'), from code that is considered commodity in the company (e.g. infrastructure, platform, and UI components), as the former most likely needs a higher level of protection than the latter.

Reply:
This is a great suggestion @bartgolsteijn ! 🚀 

I added it as its own sub-section below (see "### Split out the 'secret sauce' into separate repos"). Does this work for you?

I feel that this activity should probably done first, as it makes many of the following points easier.

Reply:
Putting it in a separate section is fine with me.

I would not put it first however, increasing security awareness and configuring the SCM system should be done first. Splitting repositories and maybe refactoring (moving around code, injecting secrets via the environment, ...) can be done later. At least in a company with many CLOSED (legacy) repositories.

Reply:
Sounds good. I will move this accordingly.